ext/pdo/pdo_stmt.c

```diff
@@ -2348,7 +2348,7 @@ PDO_API void php_pdo_free_statement(pdo_stmt_t *stmt)
 	do_fetch_opt_finish(stmt, 1);
 
 	if (!Z_ISUNDEF(stmt->database_object_handle)) {
-		zval_ptr_dtor(&stmt->database_object_handle);
+		Z_DELREF(stmt->database_object_handle);
```

Comment:
The problem as I see it is that destruction of one of the PDO objects will always rollback the transaction on the inner object (if a transaction is active). In this case this backfires because in the meantime another transaction has been opened through a different PDO object (but same inner object).